lib/modules/manager/swift/artifacts.ts

```diff
@@ -0,0 +1,87 @@
+import { quote } from 'shlex';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  getParentDir,
+  getSiblingFileName,
+  localPathExists,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+async function swiftPackageResolve(packageFile: string): Promise<void> {
+  const packagePath = getParentDir(packageFile);
+  const cmd = `swift package resolve --package-path ${quote(packagePath)}`;
+  const execOptions: ExecOptions = {
+    docker: {
+      image: 'swift',
```

Comment:
```suggestion
      image: 'sidecar',
```

swift needs to be installed as tool constraint instead,
https://github.com/renovatebot/renovate/blob/70ad9e61c1d7fe30f321d5dd1cb2a19dd20aebb4/lib/modules/manager/gradle-wrapper/artifacts.ts#L154-L160

Reply:
I started on this with e798ddbfe1f9ae8b38dc64dc6d9abe273524d1a6

The test for `binarySource: 'docker'` now fails with `Invalid tool to install: swift` - is this related to https://github.com/containerbase/base/issues/576 or should this still be working?

I should probably also add a test where `config.constraints?.swift` is set to a specific version? Are these external constraints set in any particular circumstances or with particular `binarySource: '...'` settings I should test for?

Reply:
should work in tests, check util/exec/containerbase for definitions